client/client.go

```diff
@@ -511,3 +562,116 @@ func (client *Client) DebugGet(action string, result interface{}, params map[str
 	_, err := client.doSync("GET", "/v1/debug", urlParams, nil, nil, &result)
 	return err
 }
+
+// decoder receives a raw HTTP response and performs internal client
+// processing, as well as unmarshalling the custom result.
+func (client *Client) decoder(ctx context.Context, rsp *http.Response, opts *RequestOptions, result interface{}) (*RequestResponse, error) {
```

Comment:
The decoder is the client callback mechanism that the Requester interface can use for client specific details. This allows the Requester implementation to stay clean of implementation details.